cpp/src/Slice/Parser.cpp

```diff
@@ -98,46 +98,41 @@ Slice::Metadata::Metadata(string rawMetadata, string file, int line) : GrammarBa
     _file = std::move(file);
     _line = line;
 
-    // If the metadata contains no colons, then it must be a directive with no arguments.
-    size_t firstColonPos = rawMetadata.find(':');
-    if (firstColonPos == string::npos)
-    {
-        _directive = rawMetadata;
-        _arguments = "";
-        return;
-    }
+    size_t splitPos = string::npos;
 
-    // Otherwise, we check whether the colon is for a language prefix or for arguments and split the string accordingly.
-    size_t secondColonPos = rawMetadata.find(':', firstColonPos + 1);
-    size_t splitPos;
-    if (secondColonPos == string::npos)
+    size_t firstColonPos = rawMetadata.find(':');
+    if (firstColonPos != string::npos)
     {
-        // If the metadata contains only 1 colon, we need to check if it's for a language prefix, or for arguments.
+        // Check if the metadata starts with a language prefix.
         // NOTE: It is important that this list is kept in alphabetical order!
         static const string languages[] = {"cpp", "cs", "java", "js", "matlab", "php", "python", "ruby", "swift"};
         string prefix = rawMetadata.substr(0, firstColonPos);
-        bool isLanguage = binary_search(&languages[0], &languages[sizeof(languages) / sizeof(*languages)], prefix);
-        if (isLanguage)
+        bool hasLangPrefix = binary_search(&languages[0], &languages[sizeof(languages) / sizeof(*languages)], prefix);
```

Comment:
I would convert languages to a constexpr of string_view.

Reply:
Fixed.